File: QuickStarts/CAB/CS/Walkthrough/Step4/MyModule/MyView.cs
//===============================================================================
// Microsoft patterns & practices
// Smart Client Software Factory 2010
//===============================================================================
// Copyright (c) Microsoft Corporation.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
//===============================================================================
// The example companies, organizations, products, domain names,
// e-mail addresses, logos, people, places, and events depicted
// herein are fictitious.  No association with any real company,
// organization, product, domain name, email address, logo, person,
// places, or events is intended or should be inferred.
//===============================================================================
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;
using Microsoft.Practices.CompositeUI.SmartParts;

namespace MyModule
{
	[SmartPart]
	public partial class MyView : UserControl, IMyView 
	{
		public MyView()
		{
			InitializeComponent();
		}

		#region IMyView Members

		public string Message
		{
			get
			{
				return this.label1.Text;
			}
			set
			{
				this.label1.Text = value;
			}
		}

		#endregion
	}
}

File: QuickStarts/CAB/VB/Walkthrough/Step4/MyModule/MyView.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Text
Imports System.Windows.Forms
Imports Microsoft.Practices.CompositeUI.SmartParts

Namespace MyModule
	<SmartPart()> _
	Partial Public Class MyView : Inherits UserControl : Implements IMyView

		Public Sub New()
			InitializeComponent()
		End Sub

#Region "IMyView Members"

		Public Property Message() As String Implements IMyView.Message
			Get
				Return Me.label1.Text
			End Get
			Set(ByVal value As String)
				Me.label1.Text = value
			End Set
		End Property

		Public Sub LoadHandler(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
			RaiseEvent IMyViewLoad(sender, e)
		End Sub

#End Region

		Public Event IMyViewLoad(ByVal sender As Object, ByVal e As System.EventArgs) Implements IMyView.Load
	End Class
End Namespace
